src/Orc.Controls/Orc.Controls.Shared/Controls/DateTimePicker/Views/DateTimePickerControl.xaml.cs

```diff
@@ -245,6 +294,209 @@ public override void OnApplyTemplate()
             base.OnApplyTemplate();
 
             AccentColorBrush = TryFindResource("AccentColorBrush") as SolidColorBrush;
+
+            EnableOrDisableHourConverterDependingOnFormat();
+        }
+
+        private void OnFormatChanged()
+        {
+            ApplyFormat();
+        }
+
+        private void ApplyFormat()
+        {
+            var dayFormat = Format.Count(x => x == 'd');
+            var monthFormat = Format.Count(x => x == 'M');
+            var yearFormat = Format.Count(x => x == 'y');
+            var hourFormat = 0;
+            var hour12Format = Format.Count(x => x == 'h');
+            var hour24Format = Format.Count(x => x == 'H');
+            var minuteFormat = Format.Count(x => x == 'm');
+            var secondFormat = Format.Count(x => x == 's');
+
+            if (hour12Format > 0 && hour24Format > 0)
+            {
+                throw new FormatException("Format string is incorrect. Hour field must be 12-hour or 24-hour format, but no both");
```

Comment:
Please use throw Log.ErrorAndCreateException<FormatException>("Format string is incorrect. Hour field must be 12-hour or 24-hour format, but no both");